### PR TITLE
feat: migrate AnimatedPlaceholder component (#5238

### PR DESCRIPTION
**Pull request by HinataKah0**
Migrate AnimatedPlaceholder component from nodejs.dev and create a new Story.

Related to: https://github.com/nodejs/nodejs.org/issues/5193

### CODE REVIEW

**Comment by HinataKah0** [comment]
Tested:

Screenshot 2023-04-07 at 5 22 37 PM
Screenshot 2023-04-07 at 5 23 33 PM

**Comment by mikeesto** [comment]
Thanks for your contribution! Please have a look at the lint error: https://github.com/nodejs/nodejs.org/actions/runs/4637041766/jobs/8205572868?pr=5238

**Review by rodion-arr — COMMENTED** [review]
We need to update PR description and replace `Fixes: #5193` with `Related to #5293` so it will not close entire linked issue after merge because it tracks multiple components migration

File: components/Common/AnimatedPlaceholder/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  height?: number;
+}
+
+const AnimatedPlaceholder = ({ children, width = 400, height = 50 }: Props) => (
+  <div
+    className={styles.placeholder}
+    style={{ width: `${width}px`, height: `${height}px` }}
+  >
+    {/* Prefer external skeleton structure or render default in case not passed */}
+    {children}
```
Comment (manishprivet) [inline components/Common/AnimatedPlaceholder/index.tsx]:
```suggestion
    {children || (
      <>
        <div className={styles.placeholderImage} />
        <div className={styles.placeholderText}>
          <div className={styles.placeholderTextLine} />
          <div className={styles.placeholderTextLine} />
        </div>
      </>
    )}
```

Comment (manishprivet) [reply components/Common/AnimatedPlaceholder/index.tsx]: We can probably write it like this to prevent double checks on children

Comment (HinataKah0) [reply components/Common/AnimatedPlaceholder/index.tsx]: Thanks, it looks neater! Tested as well.

**Comment by HinataKah0** [comment]
Ah,
I saw the merge commit is being overwritten.

FYI, I usually use Rebase workflow rather than adding more commits if it is atomic enough.

File: components/Common/AnimatedPlaceholder/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  width?: number;
+  height?: number;
+}
+
+const AnimatedPlaceholder = ({ children, width = 400, height = 50 }: Props) => (
+  <div
+    className={styles.placeholder}
+    style={{ width: `${width}px`, height: `${height}px` }}
+  >
+    {/* Prefer external skeleton structure or render default in case not passed */}
```
Comment (ovflowd) [inline components/Common/AnimatedPlaceholder/index.tsx]: Not sure we need an inline comment here.

File: components/Common/AnimatedPlaceholder/index.module.scss [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  background-size: 800px 104px;
+  height: 96px;
+  position: relative;
+}
+
+.placeholder {
+  display: flex;
+  width: 100%;
+
+  @keyframes placeHolderShimmer {
```
Comment (ovflowd) [inline components/Common/AnimatedPlaceholder/index.module.scss]: Can we remove the keyframes from this level? As keyframes in theory should not be contained within a class?

**Review by ovflowd — APPROVED** [review]
LGTM! Thank you for your first contribution, @HinataKah0! Left two small comments that you can address at your own discretion :)

**Review by mikeesto — APPROVED** [review]

**Review by shanpriyan — APPROVED** [review]

**Pull request merged**